datumaro/components/operations.py

```diff
@@ -768,21 +767,33 @@ def distance(self, a, b):
 class LineMatcher(_ShapeMatcher):
     @staticmethod
     def distance(a, b):
+        # Compute inter-line area by using the Trapezoid formulae
+        # https://en.wikipedia.org/wiki/Trapezoidal_rule
+        # Normalize by common bbox and get the bbox fill ratio
+        # Call this ratio the "distance"
+
         a_bbox = a.get_bbox()
         b_bbox = b.get_bbox()
         bbox = max_bbox([a_bbox, b_bbox])
-        area = bbox[2] * bbox[3]
-        if not area:
+        box_area = bbox[2] * bbox[3]
+        if not box_area:
             return 1
 
-        # compute inter-line area, normalize by common bbox
-        point_count = max(max(len(a.points) // 2, len(b.points) // 2), 5)
-        a, sa = smooth_line(a.points, point_count)
-        b, sb = smooth_line(b.points, point_count)
+        def _approx(line, segments):
+            if len(line) // 2 == segments + 1:
+                return np.reshape(line, (-1, 2))
+            else:
+                return approximate_line(line, segments=segments)
+        segments = max(max(len(a.points) // 2, len(b.points) // 2), 5) - 1
```

Comment:
The two `max` calls look redundant - `max` can take more than 2 arguments.

Reply:
Fixed